torch/distributed/_sharded_tensor/api.py

```diff
@@ -550,9 +559,46 @@ def sharding_spec(self) -> ShardingSpec:
         """
         return self._sharding_spec
 
+    def reshard(self, resharding_spec: ReshardingSpec) -> Union[ShardedTensor, torch.Tensor]:
+        """
+        Reshards a tensor given the resharding_spec, if output_local_tensors is
```

Comment:
shouldn't this be "reshard to a tensor"? Also, maybe we want a better explanation on what does it mean of "reshard to a tensor", does it mean we only output local shards, or does it mean we all_gather the local shards and make that as an output, seems a bit confusing

Reply:
Sure, let me make the comment more clear. We return the results after `all_gather`.